### PR TITLE
Optimize WelsDctT4_c and WelsDctFourT4_c with LASX

### PR DESCRIPTION
**Pull request by jinboson**
Encoding 1920x1080.yuv tested on LoongArch 3A5000 2.3GHz:
before       after
33.45fps     35.51fps

### CODE REVIEW

File: codec/encoder/core/loongarch/dct_lasx.c [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+  DUP2_ARG3(__lasx_xvpermi_q,
+            tmp1, tmp0, 0x20,
+            tmp3, tmp2, 0x20,
+            dst0, dst1);
+  DUP2_ARG3(__lasx_xvpermi_q,
+            tmp1, tmp0, 0x31,
+            tmp3, tmp2, 0x31,
+            dst2, dst3);
+  __lasx_xvst(dst0, pDct, 0);
+  __lasx_xvst(dst1, pDct + 16, 0);
```
Comment (yinshiyou) [inline codec/encoder/core/loongarch/dct_lasx.c:221]: __lasx_xvst(dst1, pDct, 16);

Same below.

File: codec/encoder/core/loongarch/dct_lasx.c [hunk 2]
```diff
@@ -0,0 +1,224 @@
+/*!
+ * \copy
+ *     Copyright (c)  2009-2018, Cisco Systems
```
Comment (yinshiyou) [inline codec/encoder/core/loongarch/dct_lasx.c:3]: should be 2022 ?

File: codec/encoder/core/loongarch/dct_lasx.c (hunk 1 above)
Comment (jinboson) [reply codec/encoder/core/loongarch/dct_lasx.c:221]: will update in next push

File: codec/encoder/core/loongarch/dct_lasx.c (hunk 2 above)
Comment (jinboson) [reply codec/encoder/core/loongarch/dct_lasx.c:3]: make sense. will update in next push

Comment (jinboson) [reply codec/encoder/core/loongarch/dct_lasx.c:3]: Nice catch, will update in next push.

**Comment by GuangweiWang** [comment]
hi @jinboson 
is it possible to provide me your e-mail address? there is some thing about this PR that I want to discuss with you.

**Comment by jinboson** [comment]
> hi @jinboson is it possible to provide me your e-mail address? there is some thing about this PR that I want to discuss with you.

my e-mail is jinbo@loongson.cn, thanks.

**Comment by yinshiyou** [comment]
LGTM.

**Comment by GuangweiWang** [comment]
Encoding 1920x1080.yuv tested on LoongArch 3A5000 2.3GHz:
before       after
33.45fps     35.51fps

**Pull request merged**